integration-libs/cdc/user-profile/register/cdc-register-component.service.spec.ts

```diff
@@ -167,18 +167,23 @@ describe('CdcRegisterComponentService', () => {
     });
 
     it('should not do anything when CDC registration fails', (done) => {
-      cdcJsService.registerUserWithoutScreenSet =
-        createSpy().and.throwError('ERROR');
+      cdcJsService.registerUserWithoutScreenSet = createSpy().and.returnValue(
+        throwError('ERROR')
+      );
 
-      cdcUserRegisterService.register(userRegisterFormData).subscribe(() => {
-        expect(connector.register).not.toHaveBeenCalled();
-        expect(cdcJsService.registerUserWithoutScreenSet).toHaveBeenCalledWith({
-          titleCode: 'Mr.',
-          firstName: 'firstName',
-          lastName: 'lastName',
-          uid: 'uid',
-          password: 'password',
-        });
+      cdcUserRegisterService.register(userRegisterFormData).subscribe({
+        error: () => {
+          expect(connector.register).not.toHaveBeenCalled();
+          expect(
+            cdcJsService.registerUserWithoutScreenSet
+          ).toHaveBeenCalledWith({
+            titleCode: 'Mr.',
+            firstName: 'firstName',
+            lastName: 'lastName',
+            uid: 'uid',
+            password: 'password',
+          });
+        },
       });
       expect(cdcJsService.didLoad).toHaveBeenCalled();
       done();
```

Comment:
this done() callback is now outside of the body of the subscription

In most cases, the Jasmine's `done` function should be put inside the body of the observable's `subscribe()` in the test, when this body contains any `expect()`.  Otherwise, Jasmine won't wait for the invocation of done  and any expect() checks placed in async operations near the done() may not be checked at all.